src/mostlyai/qa/sampling.py

```diff
@@ -67,22 +67,24 @@ def pull_data_for_accuracy(
     assert setup is None or setup in ["1:1", "1:N"]
 
     key = "__KEY"
+    max_sample_size = 10_000_000 if max_sample_size is None else max_sample_size
 
     if df_ctx is not None:
         # explicit context
-        df_ctx = df_ctx.sample(frac=1).head(max_sample_size).reset_index(drop=True)
-        df_ctx = df_ctx.rename(columns={ctx_primary_key: tgt_context_key})
-        df_tgt = df_tgt.merge(df_ctx[tgt_context_key], on=tgt_context_key).reset_index(drop=True)
+        df_ctx = df_ctx.sample(frac=1).head(max_sample_size)
+        df_ctx = df_ctx.rename(columns={ctx_primary_key: tgt_context_key}).reset_index(drop=True)
+        df_tgt = df_tgt.merge(df_ctx[tgt_context_key], on=tgt_context_key, how="inner").reset_index(drop=True)
```

Comment:
how="inner" also solves `msd-966`, right? :)

Reply:
does it? the motivation was just to make this default argument explicit, as we also set it in the following if clause

Reply:
you are right, we are doing merge with `how="left"` later on, all good then!